nixos/lib/test-driver/src/test_driver/machine/ocr.py

```diff
@@ -0,0 +1,121 @@
+import multiprocessing
+import os
+import shutil
+import subprocess
+
+from test_driver.errors import MachineError
+
+
+def perform_ocr_on_screenshot(screenshot_path: str) -> str:
+    """
+    Perform OCR on a screenshot that contains text.
+    Returns a string with all words that could be found.
+    """
+    variants = perform_ocr_variants_on_screenshot(screenshot_path, False)[0]
+    if len(variants) != 1:
+        raise MachineError(f"Received wrong number of OCR results: {len(variants)}")
+    return variants[0]
+
+
+def perform_ocr_variants_on_screenshot(
+    screenshot_path: str, variants: bool = True
+) -> list[str]:
+    """
+    Same as perform_ocr_on_screenshot but will create variants of the images
+    that can lead to more words being detected.
+    Returns a string with words for each variant.
+    """
+    if shutil.which("tesseract") is None:
+        raise MachineError("OCR requested but `tesseract` is not available")
+
+    # Tesseract runs parallel on up to 4 cores.
+    # Docs suggest to run it with OMP_THREAD_LIMIT=1 for hundreds of parallel
+    # runs. Our average test run is somewhere inbetween.
+    # https://github.com/tesseract-ocr/tesseract/issues/3109
+    processes = int(os.process_cpu_count() / 4)
```

Comment:
You probably want to have at least one process and `int(3 / 4) == 0`.

`from math import ceil`

```suggestion
    processes = ceil(os.process_cpu_count() / 4)
```